### ISSUE TITLE
Protected term errors when supporting v1 and v2

### ISSUE THREAD

**Issue by OR13** [w3c/vc-data-model#1150]
Consider:

```json
{
  "@context": ["https://www.w3.org/ns/credentials/v2"],
  "type": ["VerifiablePresentation"],
  "verifiableCredential": [
    {
      "@context": ["https://www.w3.org/2018/credentials/v1"],
      "type": ["VerifiableCredential"]
    }
  ]
}
```
> jsonld.SyntaxError: Invalid JSON-LD syntax; tried to redefine a protected term.

**Comment by mprorock** [comment]
excellent catch @OR13

**Comment by OR13** [comment]
I was able to work around this issue, by removing the v2 context from the outer object:

```ts
const multiPresentationSupport = {
        '@context': {'@vocab': 'https://www.w3.org/2018/credentials#'},
        type: ['VerifiablePresentation', 'VerifiablePresentations'],
        id: `urn:uuid:${uuid.v4()}`,
        holder: process.env.API_BASE_URL as string,
        verifiablePresentation: verifiablePresentations,
      }
```

As you can see, most of the terms that get defined related to a presentation of a presentation of a mill test report are controlled by origins other than W3C...

<img width="2702" alt="Screen Shot 2023-06-12 at 12 17 45 PM" src="https://github.com/w3c/vc-data-model/assets/8295856/771b216c-b010-4eaa-b3e8-329c50eeb679">

**Comment by OR13** [comment]
Probably a better resolution to the above error

```ts
const multiPresentationSupport = {
        '@context': {'@vocab': 'https://www.w3.org/2018/credentials#'},
        type: ['VerifiablePresentation'],
        id: `urn:uuid:${uuid.v4()}`,
        holder: holder,
        verifiableCredential: verifiablePresentations,
      }
```

<img width="2716" alt="Screen Shot 2023-06-12 at 12 28 03 PM" src="https://github.com/w3c/vc-data-model/assets/8295856/cbb1fbfc-a900-4b1d-ab70-37b527d94df2">

Notice this time the graphs are joined properly, because `https://www.w3.org/2018/credentials#verifiableCredential` is a known property of a `VerifiablePresentation` where as `verifiablePresentation` IS NOT.

**Comment by TallTed** [comment]
@OR13 — `don't run the above code`

Please put this and similar warnings *before* the code that should not be run, and optimally *break* such code making it unexecutable without some fix(es), and include some comment(s) saying that it has been broken, and *if you must* include some comment(s) explicitly describing how a bad actor could change the code to make it executable.

Note that various crawlers may act on any URLs they encounter, whether or not they're "live" on the page where they're encountered, so you may already be causing trouble for various entities.

**Comment by OR13** [comment]
@TallTed how would you suggest disabling the above code? 

Should I make the URL invalid so fetch doesn't run properly?

**Comment by OR13** [comment]
Related to https://github.com/w3c/vc-data-model/pull/1158

**Comment by OR13** [comment]
Must be done before CR, because of the context being normative.

**Comment by OR13** [comment]
@TallTed thanks! I updated the examples to throw undefined term errors that halt execution, in honor of how JSON-LD processors also throw errors when terms are not properly defined in a context.

**Comment by dlongley** [comment]
This solution to this is in [this comment](https://github.com/w3c/vc-data-model/issues/1150#issuecomment-1587797040). We should apply this when we make other context changes and rev the normative context hash.

Related PRs for potentially adding or modifying context terms:

#938
#1214
#1152
https://github.com/w3c/vc-status-list-2021/pull/69

There may be more than that. We might be able to save some effort by making the context updates at once.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-08-16-vcwg#section2-3) on 2023-08-16

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 2.3. Protected term errors when supporting v1 and v2 (issue vc-data-model#1150)

_See github issue [vc-data-model#1150](https://github.com/w3c/vc-data-model/issues/1150)._




**Kristina Yasuda:** protected term errors when doing v1 and v2, dave has pointed to multiple PRs in his latest comment, what's plan here?  

**Dave Longley:** I don't plan to do PR unless it's done as a final pass before CR over all of the other issues we have -- we have a number of things that will need tomake changes to v2 context (other specs, etc) -- we need to make pass at context at the end, tagged this item and others to make sure we get everything in there.  

**Kristina Yasuda:** waiting for v2 context to be "final final".  

**Dave Longley:** Not "final final", just right before we go to CR.  

</details>

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-08-16-vcwg#section2-3) on 2023-08-16

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 2.3. Protected term errors when supporting v1 and v2 (issue vc-data-model#1150)

_See github issue [vc-data-model#1150](https://github.com/w3c/vc-data-model/issues/1150)._




**Kristina Yasuda:** protected term errors when doing v1 and v2, dave has pointed to multiple PRs in his latest comment, what's plan here?  

**Dave Longley:** I don't plan to do PR unless it's done as a final pass before CR over all of the other issues we have -- we have a number of things that will need tomake changes to v2 context (other specs, etc) -- we need to make pass at context at the end, tagged this item and others to make sure we get everything in there.  

**Kristina Yasuda:** waiting for v2 context to be "final final".  

**Dave Longley:** Not "final final", just right before we go to CR.  

</details>

**Comment by dlongley** [comment]
This will be addressed by #1259.

**Comment by msporny** [comment]
PR #1259 has been raised to address this issue. This issue will be closed once #1259 is merged.
